cocotb/binary.py

```diff
@@ -51,6 +54,8 @@ def resolve(string):
         elif resolve_x_to == "RANDOM":
             bits = "{0:b}".format(random.getrandbits(1))
             string = string.replace(char, bits)
+        elif resolve_x_to == "CONSISTENT_RANDOM":
+            string = string.replace(char, '1' if (hash(string)*consistent_val) & 0x80000000 else '0')
```

Comment:
`resolve_x_to == "RANDOM"` is already deterministic and identical across test runs once a `RANDOM_SEED` is given. But in your case that's not sufficient, you need the *same* x-resolving behavior for all binary values?

Are there really use cases where a user passed `RANDOM_SEED` *and* still wants a different random value for each x-resolving of a binary value?
If not, making the existing branch `resolve_x_to == "RANDOM"` behave like your `resolve_x_to == "CONSISTENT_RANDOM"` would make sense.

If noone has a use case which requires the current `resolve_x_to == "RANDOM"` behavior together with a `RANDOM_SEED` I'd just to with option 1.

Reply:
I worry that with hash randomization, `hash(string)` will not give the same result across runs of cocotb